base/linalg/generic.jl

```diff
@@ -688,9 +688,18 @@ julia> dot([1; 1], [2; 3])
 
 julia> dot([im; im], [1; 1])
 0 - 2im
+
+julia> dot([[0 1; -1 0], [im 0; 0 im]], [1; 1])
+2×2 Array{Complex{Int64},2}:
+ 0-1im  -1+0im
+ 1+0im   0-1im
 ```
 """
 dot(x::AbstractVector, y::AbstractVector) = vecdot(x, y)
+dot(x::AbstractArray, y::Number) = ctranspose(x) * y
+dot(x::Number, y::AbstractArray) = conj(x) * y
+dot(x::AbstractArray, y::AbstractArray) = ctranspose(x) * y
```

Comment:
What I meant is that there already is a `dot` defined between matrices. But depending on the element type there will be different results (scalar or matrix) with this definition, which might give some unexpected behavior.